File: CS/WindowsApplication1/MyPivotGridFilterHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;
using System.Collections;
using DevExpress.XtraEditors.Controls;

namespace WindowsApplication1
{
    public class MyPivotGridFilterHelper
    {
        private PivotGridControl pivotGrid;
        private ListBoxItemCollection output;
        private ListBoxItemCollection outputDetail;


        public MyPivotGridFilterHelper(PivotGridControl pivotGrid, ListBoxItemCollection output, ListBoxItemCollection outputDetail)
        {
            this.pivotGrid = pivotGrid;
            this.output = output;
            this.outputDetail = outputDetail;
            pivotGrid.FieldFilterChanged += pivotGrid_FieldFilterChanged;
        }

        void WriteFieldFilter(PivotGridField field)
        {
            object[] values = field.FilterValues.Values;
            if (values.Length == 0) return;
            output.Add("==============");
            output.Add(String.Format("Field '{0}'. {1} values:", field.FieldName, field.FilterValues.FilterType));
            foreach (object value in values) output.Add(String.Format("[ {0} ]", value));
        }

        void WriteFieldFilterDetail(PivotGridField field)
        {
            outputDetail.Add("==============");
            outputDetail.Add(String.Format("Field '{0}'. Included values:", field.FieldName));
            foreach (object value in field.FilterValues.ValuesIncluded) outputDetail.Add(String.Format("[ {0} ]", value));
            outputDetail.Add(String.Format("Field '{0}'. Excluded values:", field.FieldName));
            foreach (object value in field.FilterValues.ValuesExcluded) outputDetail.Add(String.Format("[ {0} ]", value));
        }


        void DiplayActiveFilter()
        {
            output.Clear();
            output.Add("Active filter:");
            foreach (PivotGridField field in pivotGrid.Fields)
            {
                WriteFieldFilter(field);
            }
        }

        void DiplayActiveDetailFilter()
        {
            outputDetail.Clear();
            outputDetail.Add("Detail filter:");
            foreach (PivotGridField field in pivotGrid.Fields)
            {
                WriteFieldFilterDetail(field);
            }
        }

        void pivotGrid_FieldFilterChanged(object sender, PivotFieldEventArgs e)
        {
            DiplayActiveFilter();
            DiplayActiveDetailFilter();
        }
    }
}

File: VB/WindowsApplication1/MyPivotGridFilterHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid
Imports System.Collections
Imports DevExpress.XtraEditors.Controls

Namespace WindowsApplication1
	Public Class MyPivotGridFilterHelper
		Private pivotGrid As PivotGridControl
		Private output As ListBoxItemCollection
		Private outputDetail As ListBoxItemCollection


		Public Sub New(ByVal pivotGrid As PivotGridControl, ByVal output As ListBoxItemCollection, ByVal outputDetail As ListBoxItemCollection)
			Me.pivotGrid = pivotGrid
			Me.output = output
			Me.outputDetail = outputDetail
			AddHandler pivotGrid.FieldFilterChanged, AddressOf pivotGrid_FieldFilterChanged
		End Sub

		Private Sub WriteFieldFilter(ByVal field As PivotGridField)
			Dim values() As Object = field.FilterValues.Values
			If values.Length = 0 Then
				Return
			End If
			output.Add("==============")
			output.Add(String.Format("Field '{0}'. {1} values:", field.FieldName, field.FilterValues.FilterType))
			For Each value As Object In values
				output.Add(String.Format("[ {0} ]", value))
			Next value
		End Sub

		Private Sub WriteFieldFilterDetail(ByVal field As PivotGridField)
			outputDetail.Add("==============")
			outputDetail.Add(String.Format("Field '{0}'. Included values:", field.FieldName))
			For Each value As Object In field.FilterValues.ValuesIncluded
				outputDetail.Add(String.Format("[ {0} ]", value))
			Next value
			outputDetail.Add(String.Format("Field '{0}'. Excluded values:", field.FieldName))
			For Each value As Object In field.FilterValues.ValuesExcluded
				outputDetail.Add(String.Format("[ {0} ]", value))
			Next value
		End Sub


		Private Sub DiplayActiveFilter()
			output.Clear()
			output.Add("Active filter:")
			For Each field As PivotGridField In pivotGrid.Fields
				WriteFieldFilter(field)
			Next field
		End Sub

		Private Sub DiplayActiveDetailFilter()
			outputDetail.Clear()
			outputDetail.Add("Detail filter:")
			For Each field As PivotGridField In pivotGrid.Fields
				WriteFieldFilterDetail(field)
			Next field
		End Sub

		Private Sub pivotGrid_FieldFilterChanged(ByVal sender As Object, ByVal e As PivotFieldEventArgs)
			DiplayActiveFilter()
			DiplayActiveDetailFilter()
		End Sub
	End Class
End Namespace
